pubsub/topics/main.go

```diff
@@ -145,6 +149,37 @@ func publish(client *pubsub.Client, topic, msg string) error {
 	return nil
 }
 
+func publishThatScales(client *pubsub.Client, topic string, n int) {
+	ctx := context.Background()
+	// [START pubsub_publish_that_scales]
+	t := client.Topic(topic)
+
+	fmt.Println("\nPress ENTER to stop publishing...\n")
+
+	for i := 0; i < n; i++ {
+		result := t.Publish(ctx, &pubsub.Message{
+			Data: []byte("Message No. " + strconv.Itoa(i)),
+		})
+
+		go func(i int) {
+			// Block until the result is returned and a server-generated
+			// ID is returned for the published message.
+			id, err := result.Get(ctx)
+
+			if err != nil {
+				log.Output(1, fmt.Sprintf("Failed to publish: %v", err))
+			} else {
+				fmt.Printf("Published message No. %d; msg ID: %v\n", i, id)
+			}
+		}(i)
+
+	}
+
+	fmt.Scanln()
+	fmt.Println("You pressed ENTER to stop publishing.\n")
```

Comment:
I think using a sync.WaitGroup is the right thing to do here.
`Scanln` makes testing very difficult.

Reply:
You could also use a `context.Context` and `switch` on `ctx.Done()`.